src/Microsoft/Provider.php

```diff
@@ -58,20 +64,39 @@ protected function getTokenUrl()
      */
     protected function getUserByToken($token)
     {
-        $response = $this->getHttpClient()->get(
+        $response_user = $this->getHttpClient()->get(
             'https://graph.microsoft.com/v1.0/me',
             [
                 RequestOptions::HEADERS => [
                     'Accept'        => 'application/json',
                     'Authorization' => 'Bearer '.$token,
                 ],
                 RequestOptions::QUERY => [
-                    '$select' => implode(',', array_merge(self::DEFAULT_FIELDS, ($this->config['fields'] ?: []))),
+                    '$select' => implode(',', array_merge(self::DEFAULT_FIELDS_USER, ($this->config['user_fields'] ?: []))),
                 ],
             ]
         );
 
-        return json_decode((string) $response->getBody(), true);
+        $formatted_response = json_decode((string) $response_user->getBody(), true);
+
+        if ($this->getConfig('tenant', 'common') === 'common') {
```

Comment:
should we not make this opt in, rather than opt out?

Reply:
Have included in further changes.